internal/controllers/utils/utils.go

```diff
@@ -844,3 +846,83 @@ func GetDefaultBackendTransport() (http.RoundTripper, error) {
 
 	return net.SetTransportDefaults(&http.Transport{TLSClientConfig: tlsConfig}), nil
 }
+
+// Helper function to compare two slices of interfaces
+func IsInterfacesEqual(a, b []string) bool {
```

Comment:
we could also use https://pkg.go.dev/slices#Equal

Reply:
Changed.